botnim/benchmark/assistant_loop.py

```diff
@@ -2,60 +2,77 @@
 import json
 from pathlib import Path
 import requests_openapi
+from typing import Dict
+import re
+import logging
 
 from openai import OpenAI
 from openai.types.beta.threads.runs.run_step import ToolCallsStepDetails
 
+from botnim.query import QueryClient, run_query
+from botnim.config import get_logger, validate_environment, DEFAULT_ENVIRONMENT
 TEMP = 0
 
+logger = get_logger(__name__)
+
 def get_openapi_output(openapi_spec, tool_name, parameters):
     client = requests_openapi.Client(req_opts={"timeout": 30})
     client.load_spec_from_file(Path('specs/openapi') / openapi_spec)
-    resp = getattr(client, tool_name)(**parameters)
```

Comment:
why was this change needed?

Reply:
the purpose was to replace all that with a single call to client.call_endpoint(tool_name, parameters). reverted